pywr/hashes.py

```diff
@@ -18,19 +18,3 @@ def compute_hash(filename, algorithm="md5", chunk_size=io.DEFAULT_BUFFER_SIZE):
 
 class HashMismatchError(IOError):
     pass
-
-
-def check_hash(filename, hash, algorithm="md5", **kwargs):
```

Comment:
I wonder if some users might be relying on this a standalone  utility function? Removing it is a bit of an API change. You could keep this, but add an argument

```python
def check_hash(filename, hash, algorithm="md5", cache=None, **kwargs):
    ...
    if cache is not None:
        cache[(filename, algorithm)] = actual_hash
    
    # Then in the `Model.check_hash`
    def check_hash(self, filename, hash, algorithm="md5", **kwargs):
        ...
        check_hash(filename, hash, algorithm=algorithm, cache=self._file_hashes, **kwargs)
```

```